nautobot/dcim/views.py

```diff
@@ -1189,6 +1290,202 @@ class DeviceBayTemplateBulkDeleteView(generic.BulkDeleteView):
+                                except re.error:
+                                    obj.new_position = obj.position
+                            else:
+                                obj.new_position = obj.position.replace(find, replace)
+                            renamed_pks.append(obj.pk)
+
+                        if "_apply" in request.POST:
+                            for obj in selected_objects:
+                                obj.position = obj.new_position
+                                obj.save()
+
+                            # Enforce constrained permissions
+                            if self.queryset.filter(pk__in=renamed_pks).count() != len(selected_objects):
+                                raise ObjectDoesNotExist
+
+                            messages.success(
+                                request,
+                                f"Renamed {len(selected_objects)} {self.queryset.model._meta.verbose_name_plural}",
+                            )
+                            return redirect(self.get_return_url(request))
+
+                except ObjectDoesNotExist:
+                    msg = "Object update failed due to object-level permissions violation"
+                    self.logger.debug(msg)
+                    form.add_error(None, msg)
+
+        else:
+            form = _Form(initial={"pk": query_pks})
+
+        return Response(
+            {
+                "template": "dcim/modulebay_bulk_rename.html",
+                "form": form,
+                "obj_type_plural": self.queryset.model._meta.verbose_name_plural,
+                "selected_objects": selected_objects,
+                "return_url": self.get_return_url(request),
+                "parent_name": self.get_selected_objects_parents_name(selected_objects),
+            }
+        )
+
+
+class ModuleBayTemplateUIViewSet(
+    ModuleBayCommonViewSetMixin,
+    ObjectEditViewMixin,
+    ObjectDestroyViewMixin,
+    ObjectBulkDestroyViewMixin,
+    ObjectBulkUpdateViewMixin,
+):
+    queryset = ModuleBayTemplate.objects.all()
+    filterset_class = filters.ModuleBayTemplateFilterSet
+    bulk_update_form_class = forms.ModuleBayBulkEditForm
+    create_form_class = forms.ModuleBayTemplateCreateForm
+    form_class = forms.ModuleBayTemplateForm
+    model_form_class = forms.ModuleBayTemplateForm
+    serializer_class = serializers.ModuleBayTemplateSerializer
+    table_class = tables.ModuleBayTemplateTable
+    create_template_name = "dcim/device_component_add.html"
+
+    def get_selected_objects_parents_name(self, selected_objects):
+        selected_object = selected_objects.first()
+        if selected_object:
+            parent = selected_object.device_type or selected_object.module_type
+            return parent.display
+        return ""
+
+    @action(detail=False, methods=["POST"], url_path="rename", url_name="bulk_rename")
+    def bulk_rename(self, request, *args, **kwargs):
+        return self._bulk_rename(request, *args, **kwargs)
```

Comment:
Curious why this indirection is necessary - why not just define `bulk_rename` directly on `ModuleBayCommonViewSetMixin`?

Reply:
I don't think the function decorator works properly for subclasses. At least that's what I was experiencing which is why I did it this way.